handler/src/main/java/io/netty/handler/ssl/JdkSslContext.java

```diff
@@ -380,9 +380,7 @@ private SSLEngine configureAndWrapEngine(SSLEngine engine, ByteBufAllocator allo
             return ((JdkApplicationProtocolNegotiator.AllocatorAwareSslEngineWrapperFactory) factory)
                     .wrapSslEngine(engine, alloc, apn, isServer());
         }
-        if (endpointIdentificationAlgorithm != null) {
-            engine.getSSLParameters().setEndpointIdentificationAlgorithm(endpointIdentificationAlgorithm);
-        }
+        engine.getSSLParameters().setEndpointIdentificationAlgorithm(endpointIdentificationAlgorithm);
```

Comment:
It's not easy to come up with a test for this use-case, because the default for JDK provider is `null` today